src/decode.rs

```diff
@@ -3525,280 +3525,233 @@ unsafe fn decode_b(
 
         // angle delta
         if b_dim[2] + b_dim[3] >= 2
-            && b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int >= VERT_PRED as libc::c_int
-            && b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int
-                <= VERT_LEFT_PRED as libc::c_int
+            && b.y_mode() as IntraPredMode >= VERT_PRED
+            && b.y_mode() as IntraPredMode <= VERT_LEFT_PRED
         {
-            let acdf: *mut uint16_t =
-                (ts.cdf.m.angle_delta[(b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int
-                    - VERT_PRED as libc::c_int) as usize])
-                    .as_mut_ptr();
-            let angle =
-                dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf, 6 as libc::c_int as size_t)
-                    as libc::c_int;
-            b.c2rust_unnamed.c2rust_unnamed.y_angle = (angle - 3) as int8_t;
+            let acdf = &mut ts.cdf.m.angle_delta[b.y_mode() as usize - VERT_PRED as usize];
+            let angle = dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf.as_mut_ptr(), 6);
+            *b.y_angle_mut() = angle as int8_t - 3;
```

Comment:
Ah, I think it's this one that's changed slightly.  I'm not sure of the range of `angle`, but if `angle` ends up being `128`, then `angle as i8 - 3 == -4`, while `(angle - 3) as i8 == 125`.

Reply:
`angle` should be in the range 0 to 5 inclusive

Reply:
In that case, it doesn't really matter for this case, but I've seen similar cases to these break due to reordering arithmetic and casts, so we should generally try to stick closely to the C.

Reply:
That's a good point, we should probably stick to the original ordering here. Fixed.

Reply:
Oh nope, that doesn't actually work. In the original logic `angle` is cast to a `c_int`, THEN we do `angle - 3`. When I rewrote it I removed the intermediate cast to `c_int`, and instead cast it directly to `i8`. So what I had was correct, I had just forgot why 😅

Reply:
Oh yeah, I forgot about the C implicit `int` conversions.  It should be all good here now, as `angle` is in `0..=5` anyways.